_posts/2016/02/2016-02-03-swdcretreat2015-university_course.html

```diff
@@ -0,0 +1,437 @@
+  The notebook annotations are aimed to help those who are using the book as instructors and/or students.
+</p>
+
+<p>
+  <h4>
+    A response to my original discuss question revolved around administrative and university department politics on who should and who may teach a course like this.
+    For example, some people mentioned that the computer science department will push back to any course involving programming.
+    How did you over come these hurdles, or any others?
+  </h4>
+</p>
+
+<p>
+  [T] This was not much of an issue for me because Quest is a liberal arts college, and this is essentially the only computer science course they are offering this year.
+</p>
+
+<p>
+  [J] I just sneak it into courses that are already on the books.
+  I have created a course from scratch, and it was a nightmare.
+  My advice is to teach it in courses that already have a number (i.e., approved by the university).
+  But that's not a longterm solution.
+  It is better if you pilot it in an existing course or a topics course.
+</p>
+
+<p>
+  <h4> How would other people go about piloting a course like this?
+    For example, I'm just a graduate student, would me getting a faculty to sponsor to run a special topics course be the only way to get started?
+  </h4>
+</p>
+
+<p>
+  [J] If someone at UBC wanted to offer this, and isn't me (with teaching something or with rights),
+  I would suggest to teach it as a variable special topics.
+  The other option is to find a course that is related, and offer to teach a lap with them.
+  You can sell the SWC stuff as a lab to an an existing course.
+  This would also provide the added benefit of getting you paid as a TA, for example.
+</p>
+
+<p>
+  <h4>
+    How do we organize the SWC material into a 30 week course?
+    Should we expand the current material into 30 weeks, or offer 2 half semester courses?
+    Bill Mills expressed a concern that a 30 week course could too easily devolve into a Machine Learning/Data science course because instructors overshoot what can be crammed into a computing course.
+    SWC tries to teach fundamental skills.
+    How do we break up the current material?
+  </h4>
+</p>
+
+<p>
+  [T] My current course is taught as a 3.5 week course.
+  This is supposed to be the same number of hours as a full semester course.
+  We cover an abbreviated shell and 6 hours of version control.
+  Since code review was part of the course, learning to send PRs is essential.
+  These sessions were split up a week apart.
+  Then we covered all the Python basics.
+  Data munging and visualizations are really important, so we spend a lot of time with Visualizations and Pandas.
+  Because I think you learn a lot by doing, almost the last 9 hours of the course are done doing hack sessions, where I am around to help.
+  Quest also prefers courses not to be lecture based to promote active learning, so I'm not teaching during the hack sessions.
+</p>
+
+<p>
+  [J] I'm not teaching classic SWC material
```

Comment:
Pls replace the sentences prior to "One of ..." with: I don't necessarily teach classic SWC material in the same format used in two-day workshops. I incorporate it as we go through an existing course.